src/linklabel.rs

```diff
@@ -32,7 +32,7 @@ pub enum ReferenceLabel<'a> {
 pub type LinkLabel<'a> = UniCase<CowStr<'a>>;
 
 /// Returns number of bytes (including brackets) and label on success.
-pub(crate) fn scan_link_label<'a>(text: &'a str) -> Option<(usize, ReferenceLabel<'a>)> {
+pub(crate) fn scan_link_label(text: &str) -> Option<(usize, ReferenceLabel)> {
```

Comment:
Using an anonymous lifetime like `ReferenceLabel<'_>` may be clearer here. There is actually the allow-by-default lint for this in the compiler (`elided_lifetime_in_path`) which is planned to be warn-by-default some time in the future iirc.

Reply:
I recall reading something about this. Good catch, will change!